packages/react-tooltip/src/components/Tooltip/Tooltip.tsx

```diff
@@ -3,12 +3,11 @@ import { useTooltip } from './useTooltip';
 import { renderTooltip } from './renderTooltip';
 import { useTooltipStyles } from './useTooltipStyles';
 import type { TooltipProps } from './Tooltip.types';
-import type { ForwardRefComponent } from '@fluentui/react-utilities';
 
 /**
  * A tooltip provides light weight contextual information on top of its target element.
  */
-export const Tooltip: ForwardRefComponent<TooltipProps> = React.forwardRef((props, ref) => {
+export const Tooltip = React.forwardRef<HTMLElement, TooltipProps>((props, ref) => {
```

Comment:
Since we don't have a root slot, this probably shouldn't use forwardRef. If someone needs a ref to the content slot, they can pass it as an attribute to that slot.

Reply:
Well, it should as we also forwarding refs for composition purpose to trigger:

```tsx
<MenuTrigger>
  <Tooltip>
    <button />
  </Tooltip>
</MenuTrigger>
```

In this example `MenuTrigger` still need to pass `ref` to `button`, without ref forwarding it's not possible. As an option we can use cast to a regular `React.FunctionComponent` to hide this contract. WDYT?

Reply:
Just to make sure I understand, if I write the following, you're saying `myRef` be the `<button />`? 
```jsx
<Tooltip ref={myRef} content="...">
   <button />
</Tooltip>
```
That seems weird to me... first that the ref on Tooltip goes to another element other than the Tooltip, and secondly if we're forwarding `ref`, why aren't we forwarding other props?

----

Stepping back a minute, I wonder if another way to solve the problem of applying props to the trigger would be to update `onlyChild` to know to drill through to the children of `Tooltip` and `MenuTrigger` the same way that it currently drills through to the children of `React.Fragment`. That way both MenuTrigger and Tooltip would be able to "see through" each other to the trigger element, and they could apply props and refs to the trigger directly, instead of relying on the other to forward them.

E.g. maybe we could have an attribute e.g. `isFluentTrigger` (except with a better name) on the Tooltip and MenuTrigger types like:
```ts
// in Tooltip.tsx:
Tooltip.isFluentTrigger = true;

// in MenuTrigger.tsx:
MenuTrigger.isFluentTrigger = true;

// in onlyChild.ts:
const isFluentTrigger = <P>(type: React.ReactElement<P>['type'] & { isFluentTrigger?: boolean }) => {
  return typeof type !== 'string' && 'isFluentTrigger' in type && type.isFluentTrigger;
};

// Then onlyChild would check isFluentTrigger in addition to checking for React.Fragment
```

Reply:
> Just to make sure I understand, if I write the following, you're saying `myRef` be the `<button />`?
> 
> ```js
> <Tooltip ref={myRef} content="...">
>    <button />
> </Tooltip>
> ```
> 
> That seems weird to me... first that the ref on Tooltip goes to another element other than the Tooltip, and second if we're forwarding `ref`, why aren't we forwarding other props?

Correct and we are forwarding props. It's how composition of triggers works.

Reply:
> Stepping back a minute, I wonder if another way to solve the problem of applying props to the trigger would be to update `onlyChild` to know to drill through to the children of `Tooltip` and `MenuTrigger` the same way that it currently drills through to the children of `React.Fragment`. That way both MenuTrigger and Tooltip would be able to "see through" each other to the trigger element, and they could apply props and refs to the trigger directly, instead of relying on the other to forward them.

We had similar approach in N*, but with React you cannot predict element's children:

```tsx
function MyTrigger(props) {
  return <MenuTrigger {...props} />
}
```

In this example `MyTrigger.isFluentTrigger` will be `undefined`, but it still valid code.